src/senaite/ast/view.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of SENAITE.AST.
+#
+# SENAITE.AST is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free
+# Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2020-2022 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+from bika.lims import _
+from bika.lims import api
+from bika.lims.api import get_object
+from bika.lims.api import get_portal_type
+from bika.lims.catalog import SETUP_CATALOG
+from Products.CMFPlone.utils import safe_unicode
+from Products.Five import BrowserView
+from transaction import savepoint
+
+
+class DuplicateView(BrowserView):
+    """Displays a list of object in a table to duplicate.
+    """
+    created = []
+
+    def copy_object(self, source, title):
+        """Creates a copy of the given object, but with the given title
+        """
+        # Validate the title
+        portal_type = get_portal_type(get_object(source))
+        err_msg = self.check_title(portal_type, title)
+
+        if err_msg:
+            self.context.plone_utils.addPortalMessage(
+                safe_unicode(err_msg), "error")
+            return
+
+        # Create a copy
+        return api.copy_object(source, title=title, ShortTitle="")

```

Comment:
You can remove `ShortTitle`, cause this field belongs to `Analysis`-like objects only:

```python
return api.copy_object(source, title=title)
```